Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+        Eyelink('StopRecording'); % Stop tracker recording
+                
+        % STEP 5.7: CREATE VARIABLES FOR DATAVIEWER; END TRIAL
+        
+        % Write !V TRIAL_VAR messages to EDF file: creates trial variables in DataViewer
+        % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Trial Message Commands
+        Eyelink('Message', '!V TRIAL_VAR iteration %d', i); % Trial iteration
+        Eyelink('Message', '!V TRIAL_VAR image %s', imgName); % Image name
+        WaitSecs(0.001); % Allow some time between messages. Some messages can be lost if too many are written at the same time
+        Eyelink('Message', '!V TRIAL_VAR fix_completed %s', fixWinComplete); % Was gaze maintained inside fixation window successfully (yes/no)?        
+        Eyelink('Message', '!V TRIAL_VAR rt %d', reactionTime); % Key press RT [ms] from stimulus onset                
+        % Write TRIAL_RESULT message to EDF file: marks the end of a trial for DataViewer
+        % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Defining the Start and End of a Trial
+        Eyelink('Message', 'TRIAL_RESULT 0');
+        WaitSecs(0.01); % Allow some time before ending the trial
+        
+        % Clear Screen() textures that were initialized for each trial iteration
+        Screen('Close', backgroundTexture);
+        Screen('Close', imgTexture);
+        Screen('Close', stimTexture);       
+    end % End trial loop
+    
+    
+    %% STEP 6: CLOSE EDF FILE. TRANSFER EDF COPY TO DISPLAY PC. CLOSE EYELINK CONNECTION. FINISH UP
+    
+    % Put tracker in idle/offline mode before closing file. Eyelink('SetOfflineMode') is recommended.
+    % However if Eyelink('Command', 'set_idle_mode') is used, allow 50ms before closing the file as shown in the commented code:
+    % Eyelink('Command', 'set_idle_mode');% Put tracker in idle/offline mode
+    % WaitSecs(0.05); % Allow some time for transition 
+    Eyelink('SetOfflineMode'); % Put tracker in idle/offline mode
+    Eyelink('Command', 'clear_screen 0'); % Clear Host PC backdrop graphics at the end of the experiment
+    WaitSecs(0.5); % Allow some time before closing and transferring file    
+    Eyelink('CloseFile'); % Close EDF file on Host PC       
+    % Transfer a copy of the EDF file to Display PC
+    transferFile; % See transferFile function below  
+catch % If syntax error is detected
+    cleanup;
+    % Print error message and line number in Matlab's Command Window
+    psychrethrow(psychlasterror);
+end
+
+% Function that determines if gaze x y coordinates are within fixation window
+    function fix = inFixWindow(mx,my)        
+        fix = mx > fixationWindow(1) &&  mx <  fixationWindow(3) && ...
+            my > fixationWindow(2) && my < fixationWindow(4) ;
+    end
+
+% Cleanup function used throughout the script above
+    function cleanup
+        try

```

Comment:
``sca;`` is often a nice shorthand to clean up Screen related stuff, e.g., it closes the windows, shows hidden cursors and a bunch of other cleanup stuff. Generally most PTB demo code nowadays uses sca;

Reply:
resolved in 0abf0cbe36691edea3a6d49a87b79149ad7e6f94